homeassistant/config_entries.py

```diff
@@ -1296,6 +1296,28 @@ async def async_step_dhcp(
         """Handle a flow initialized by DHCP discovery."""
         return await self.async_step_discovery(discovery_info)
 
+    @callback
+    def async_create_entry(  # pylint: disable=arguments-differ
```

Comment:
Maybe a separate name with _option to avoid the pylint exception?

Reply:
I think this is a compatible, transparent way of overloading, with a compatible signature for existing code.

IMHO pylint isn't wrong in this case.

Reply:
I've had to work around this before too. I wish mypy could detect that it's just optional extensions, but meh..

I just feel it can be somewhat more error probe to disable linting. There is always a risk of somebody now feels like adding another non optional/non compatible arg to the function. They won't get the heads up on the fault.